spec/ysh-stdlib-synch.test.sh

```diff
@@ -0,0 +1,56 @@
+# spec/ysh-stdlib
+
+## our_shell: ysh
+
+#### semaphore
+source --builtin draft-synch.ysh
+
+sema-new (1, &s)
+fork { 
+  sleep 0.5
+  sema-down (s)
+  echo 1
+}
+fork { 
+  sleep 1
+  sema-down (s)
+  echo 2
+}
+fork { 
+  sleep 1.5
```

Comment:
We should reduce these sleep values when we add more test cases

Because I don't want to make the suite slow